docs/notifications.md

```diff
@@ -0,0 +1,122 @@
+# Notifications
+
+Key to workflows is receiving timely notifications in the places where you work. This module provides a developer framework for creating notifications, either as hardcoded business logic or by making events, messages, recipients and destinations available to the Workflow builder UI.
+
+This component is powered by the [Workflows plugin](https://github.com/humanmade/Workflows). You can find more [detailed documentation on the plugin's wiki](https://github.com/humanmade/Workflows/wiki).
+
+## Basic hardcoded example
+
+The simplest form of a notification looks like the following:
+
+```php
+HM\Workflows\Workflow::register( 'notify_editor_on_new_post' )
+  // The action hook to trigger on.
+  ->when( 'draft_to_pending' )
+  // The message to be sent. Can also be a callback that recieves data from the action hook.
+  ->what( 'A new post is ready for review' )
+  // The user role to notify. Could also be an array of user IDs.
+  ->who( 'editor' )
+  // Use the 'email' destination. Can be used multiple times for multiple destinations.
+  ->where( 'email' );
```

Comment:
This API is bonkers as hell! :D

Reply:
Yeah! Didn't think you'd be a fan 😄 Now I've had time to reflect and build on it I would like to go back to the drawing board really.

There's definite merit in the `Event` and `Destination` object implementation at least. `Destination` could really be any sort of follow on action though, not necessarily a notification.

Reply:
Yeah, I love the functionality :D